### PR TITLE
CORE-19290 - Added configuration field `tokenCacheExpiryPeriodMilliseconds`

### PR DESCRIPTION
**Pull request by filipesoliveira**

### CODE REVIEW

File: data/config-schema/src/main/java/net/corda/schema/configuration/LedgerConfig.java [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+
+    public static final String UTXO_TOKEN_MIN_DELAY_BEFORE_NEXT_FULL_SYNC = "tokens.minDelayBeforeNextFullSync";
+    public static final String UTXO_TOKEN_MIN_DELAY_BEFORE_NEXT_PERIODIC_SYNC = "tokens.minDelayBeforeNextPeriodicSync";
+    public static final String UTXO_TOKEN_FULL_SYNC_BLOCK_SIZE = "tokens.fullSyncBlockSize";
+    public static final String UTXO_TOKEN_PERIODIC_CHECK_BLOCK_SIZE = "tokens.periodCheckBlockSize";
+    public static final String UTXO_TOKEN_SEND_WAKEUP_MAX_RETRY_ATTEMPTS = "tokens.sendWakeUpMaxRetryAttempts";
+    public static final String UTXO_TOKEN_SEND_WAKEUP_MAX_RETRY_DELAY = "tokens.sendWakeUpMaxRetryDelay";
+    public static final String UTXO_TOKEN_CACHED_TOKEN_PAGE_SIZE = "tokens.cachedTokenPageSize";
+    public static final String UTXO_TOKEN_CLAIM_TIMEOUT_SECONDS = "tokens.claimTimeoutSeconds";
+    public static final String UTXO_TOKEN_CACHE_REFRESH_PERIOD_MILLISECONDS = "tokens.tokenCacheRefreshPeriodMilliseconds";
```
Comment (vkolomeyko) [inline data/config-schema/src/main/java/net/corda/schema/configuration/LedgerConfig.java]:
```suggestion
    public static final String UTXO_TOKEN_CACHE_REFRESH_PERIOD_MILLISECONDS = "tokens.tokenCacheExpiryPeriodMilliseconds";

```

File: data/config-schema/src/main/resources/net/corda/schema/configuration/ledger.utxo/1.0/corda.ledger.utxo.json [hunk 2]
```diff
@@ -1,78 +1,85 @@
...
+        },
+        "claimTimeoutSeconds": {
+          "description": "Time in seconds before the claim will be automatically released (default 10 minutes)",
+          "type": "integer",
+          "minimum": 1,
+          "default": 600
+        }
+      },
+        "tokenCacheRefreshPeriodMilliseconds": {
+          "description": "Period in milliseconds in which the token cache is refreshed (default 200 milliseconds)",
```
Comment (vkolomeyko) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/ledger.utxo/1.0/corda.ledger.utxo.json]:
```suggestion
          "description": "Expiry period in milliseconds for token in the cache (default 200 milliseconds)",

```

File: data/config-schema/src/main/resources/net/corda/schema/configuration/ledger.utxo/1.0/corda.ledger.utxo.json [hunk 3]
```diff
@@ -1,78 +1,85 @@
...
+          "default": 1500
+        },
+        "claimTimeoutSeconds": {
+          "description": "Time in seconds before the claim will be automatically released (default 10 minutes)",
+          "type": "integer",
+          "minimum": 1,
+          "default": 600
+        }
+      },
+        "tokenCacheRefreshPeriodMilliseconds": {
```
Comment (vkolomeyko) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/ledger.utxo/1.0/corda.ledger.utxo.json]:
```suggestion
        "tokenCacheExpiryPeriodMilliseconds": {

```

File: data/config-schema/src/main/resources/net/corda/schema/configuration/ledger.utxo/1.0/corda.ledger.utxo.json [hunk 4]
```diff
@@ -1,78 +1,85 @@
...
+          "description": "Period in milliseconds in which the token cache is refreshed (default 200 milliseconds)",
+          "type": "long",
+          "minimum": 0,
```
Comment (vkolomeyko) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/ledger.utxo/1.0/corda.ledger.utxo.json]: We probably need to document what value of `0` would mean:
- expiry instantly;
or
- never expires.

File: data/config-schema/src/main/java/net/corda/schema/configuration/LedgerConfig.java (hunk 1 above)
Comment (filipesoliveira) [reply data/config-schema/src/main/java/net/corda/schema/configuration/LedgerConfig.java]: The suggestion has been merged.

File: data/config-schema/src/main/resources/net/corda/schema/configuration/ledger.utxo/1.0/corda.ledger.utxo.json (hunk 3 above)
Comment (filipesoliveira) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/ledger.utxo/1.0/corda.ledger.utxo.json]: This change has been merged.

File: data/config-schema/src/main/resources/net/corda/schema/configuration/ledger.utxo/1.0/corda.ledger.utxo.json (hunk 2 above)
Comment (filipesoliveira) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/ledger.utxo/1.0/corda.ledger.utxo.json]: This change has been merged.

File: data/config-schema/src/main/resources/net/corda/schema/configuration/ledger.utxo/1.0/corda.ledger.utxo.json (hunk 4 above)
Comment (filipesoliveira) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/ledger.utxo/1.0/corda.ledger.utxo.json]: The description was updated to document the meaning of a zero value for `tokenCacheExpiryPeriodMilliseconds` config field.

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**